aerogear/origin-web-console
# [AEROGEAR-2568] Allow the DMZ URL to be set in the MAR view

## Description
Allow the DMZ Url for a mobile client to be set in the MAR view. This is added under the mobile client details section of the MAR view and should be able to be edited inline. Also add an `Edit YAML` option under the actions list in order to allow the DMZ Url to be edited within the YAML file. 

![peek 2018-05-11 12-30](https://user-images.githubusercontent.com/9078522/39922591-b277f764-5517-11e8-86ac-c8ae18f0ec9f.gif)

## Progress
- [x] Add a DMZ Url field under the `Mobile Client Details` section of the MAR view
- [x] Allow the DMZ Url to be edited inline
- [x] Add `Edit YAML` option under the actions list within the MAR view

## Verification Steps
**NOTE:** Requires changes from this [PR](https://github.com/openshift/origin-web-common/pull/320)

1. Provision a mobile application.
2. Go into the mobile client view
3. Verify that a field `DMZ Url` exists under the `Mobile Client Details` section.
4. Edit the DMZ url by clicking on the edit button
5. Enter an invalid url
6. Verify that an error message appears saying that the url provided is not valid and the save button is disabled.
7. Clear the input by clicking on the (x) button within the input field.
8. Set the DMZ url with a valid url
9. Verify that the DMZ url is successfully set. (The mobile client's `spec.dmzUrl` should be defined)
10. Verify that an `Edit YAML` option is available under the `Actions` list
11. Verify that the `spec.dmzUrl` property reflects what's shown in the `DMZ Url` field under the `Mobile Client Details`
12. Verify that no notification appears when the DMZ url is edited with no changes.

## Additional Notes
Related JIRA Ticket - https://issues.jboss.org/browse/AEROGEAR-2568
